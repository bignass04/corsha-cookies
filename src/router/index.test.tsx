import React from 'react'
import '@testing-library/jest-dom'
import { render, act, screen } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'
import Router from '.'

vi.mock('mocks/browser', () => ({
  browser: {
    start: vi.fn()
  }
}))

describe('Router', () => {
  test('renders users on the page', async () => {
    let comp
    act(() => {
      comp = render(
        <BrowserRouter>
          <Router />
        </BrowserRouter>
      )
    })
    expect(await screen.findByLabelText('Choose Cookies', undefined, { timeout: 3000 })).toBeInTheDocument()
    expect(comp.container).toMatchSnapshot()
  })
})
