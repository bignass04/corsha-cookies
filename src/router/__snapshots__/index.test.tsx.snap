// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Router > renders users on the page 1`] = `
<div>
  <main>
    <form>
      <div
        class="field-wrapper"
      >
        <label>
          Choose Cookies
          <select
            name="cookieType"
          >
            <option
              value="chocolate"
            >
              Chocolate Chip
            </option>
            <option
              value="shortbread"
            >
              Shortbread
            </option>
            <option
              value="macaroon"
            >
              Macaroon
            </option>
          </select>
        </label>
        <label>
          Quantity
          <select
            name="cookieQuantity"
          >
            <option
              value="1"
            >
              1
            </option>
            <option
              value="2"
            >
              2
            </option>
            <option
              value="3"
            >
              3
            </option>
            <option
              value="4"
            >
              4
            </option>
            <option
              value="5"
            >
              5
            </option>
          </select>
        </label>
      </div>
      <div
        class="button-wrapper"
      >
        <button>
          Add Cookies
        </button>
        <button
          class="second-button"
        >
          Remove Cookies
        </button>
      </div>
    </form>
    <section
      class="cookie-display"
    >
      <div>
        <span>
          Chocolate Chip:
        </span>
        <span
          data-testid="chocolateChip"
        >
          0
        </span>
      </div>
      <div>
        <span>
          Shortbread:
        </span>
        <span
          data-testid="shortbread"
        >
          0
        </span>
      </div>
      <div>
        <span>
          Macaroon:
        </span>
        <span
          data-testid="macaroon"
        >
          0
        </span>
      </div>
    </section>
  </main>
</div>
`;
